### DISCUSSION TITLE
Is it possible to switch from a TG to another TG without changing the page we are in ? (i use a MIDI controller)

### DISCUSSION THREAD

**Discussion by Onosendaifr** [probonopd/MiniDexed#677] [Q&A]
With the rotary knob and GPIO it seems to be possible, but i use a MIDI controller with MIDI notes to control my Mini Dexed menu, and everytime i switch TG, the first page always come back to Voice by default.

**Comment by probonopd** [comment]
Can you please describe your setup, ini file, and the exact steps you are doing. What should happen, and what does happen instead?

**Comment by Onosendaifr** [comment]
My setup : Mini Dexed + Midi Fighter 3D (simple MIDI controller that send MIDI notes)

Steps : 

- I select TG1 > Pan
- I tweak Pan
- I want to stay in the same page but select another TG
- I press the "TG up" button on my MIDI controller (i've setup everything before in the .ini file)
- Mini Dexed switch to TG2, but instead of staying in the same page, come back to the Home/Voice menu

Switching from one TG to another should not reset the current menu position

**Reply by probonopd** [reply]
Check out
https://github.com/probonopd/MiniDexed/wiki/Menu#shortcuts

**Reply by Onosendaifr** [reply]
I saw that, but it seems to work only with a knob ? I use MIDI button, how can i press and rotate

**Reply by probonopd** [reply]
I see. I don't think we have a similar shortcut implemented for this situation yet.

**Reply by Onosendaifr** [reply]
You should add it (if possible) for those who don't use a rotary encoder. 

Pressing MIDI button TG up or down should not change the page we are in, just showing a quick TG number in the upper line to confirm the change, it will fix so much the actual menu diving !
